addons/website_slides/static/src/js/slides_join_channel.js

```diff
@@ -0,0 +1,81 @@
+odoo.define('website_slides.slides_join_channel', function (require) {
+'use strict';
+
+var core = require('web.core');
+var Widget = require('web.Widget');
+var sAnimations = require('website.content.snippets.animation');
+require('website_slides.slides');
+
+var _t = core._t;
+
+var JoinChannelButton = Widget.extend({
+    events: {
+        'click .o_wslides_join_channel_link': '_onClickJoin',
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {Object} $el
+     * @param {String} message
+     */
+    _popoverAlert: function ($el, message) {
+        $el.popover({
+            trigger: 'focus',
+            placement: 'bottom',
+            container: 'body',
+            html: true,
+            content: function () {
+                return message;
+            }
+        }).popover('show');
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     */
+    _onClickJoin: function (event) {
+        var channelId = $(event.currentTarget).data('channel-id');
+        var self = this;
+        this._rpc({
+            route: '/slides/channel/join',
+            params: {
+                channel_id: channelId,
+            },
+        }).then(function (data) {
+            if (! data.error) {
+                self.$el.find('a.o_wslides_join_channel_link').text('Joined');
```

Comment:
Not sure but I think you need to make 'Joined' translatable somehow? With "_t()" ?

Reply:
And I think you can use self.$('a.o_wslides_join_channel_link') as a shortcut instead of self.$el.find()

Reply:
Right !

Reply:
Right (bis)